cpp/velox/memory/VeloxMemoryPool.cc

```diff
@@ -373,33 +396,36 @@ class WrappedVeloxMemoryPool final : public velox::memory::MemoryPool {
     visitor(subtreeMemoryUsage_);
   }
 
-  int64_t cap_;
-  velox::memory::MemoryManager& memoryManager_;
+  const std::shared_ptr<velox::memory::MemoryUsageTracker> memoryUsageTracker_;
+  velox::memory::MemoryManager* const memoryManager_;
+  velox::memory::MemoryAllocator* const allocator_;
+  const DestructionCallback destructionCb_;
 
   // Memory allocated attributed to the memory node.
   velox::memory::MemoryUsage localMemoryUsage_;
-  std::shared_ptr<velox::memory::MemoryUsageTracker> memoryUsageTracker_;
   mutable folly::SharedMutex subtreeUsageMutex_;
   velox::memory::MemoryUsage subtreeMemoryUsage_;
-  velox::memory::MemoryAllocator& allocator_;
   std::shared_ptr<VeloxMemoryAllocatorVariant> glutenAllocator_;
 };
 
 std::shared_ptr<velox::memory::MemoryPool> AsWrappedVeloxMemoryPool(gluten::MemoryAllocator* allocator) {
   return std::make_shared<WrappedVeloxMemoryPool>(
-      velox::memory::MemoryManager::getInstance(),
-      "wrapped",
+      &velox::memory::MemoryManager::getInstance(),
+      "wrappedRoot",
+      velox::memory::MemoryPool::Kind::kAggregate,
+      nullptr,
       nullptr,
       std::make_shared<VeloxMemoryAllocatorVariant>(allocator));
 }
 
-velox::memory::MemoryPool* GetDefaultWrappedVeloxMemoryPool() {
-  static WrappedVeloxMemoryPool default_pool(
-      velox::memory::MemoryManager::getInstance(),
-      "root",
+std::shared_ptr<velox::memory::MemoryPool> GetDefaultWrappedVeloxMemoryPool() {
+  return std::make_shared<WrappedVeloxMemoryPool>(
```

Comment:
Maybe this one causes the crash. Please try with a fix.

Reply:
If we make it static. it will cause bad_weak_ptr error. (I already try before.)

Reply:
> Maybe this one causes the crash. Please try with a fix.

Thanks, change it to static will fix core. But it doesn't seem very standard.
because If we do not use addChild to create pool.  we will get another core.

```
E0329 17:51:21.130357 1848225 Exceptions.h:68] Line: ../../velox/common/memory/MemoryPool.cpp:138, Function:dropChild, Expression: ret == 1 (0 vs. 1) Child memory pool root doesn't exist in Memory Pool[__default_root__ AGGREGATE MALLOC], Source: RUNTIME, ErrorCode: INVALID_STATE
terminate called after throwing an instance of 'facebook::velox::VeloxRuntimeError'
  what():  Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: (0 vs. 1) Child memory pool root doesn't exist in Memory Pool[__default_root__ AGGREGATE MALLOC]
Retriable: False
Expression: ret == 1
Function: dropChild
File: ../../velox/common/memory/MemoryPool.cpp
Line: 138
Stack trace:

```

Reply:
You can make it

```
static std::shared_ptr<WrappedVeloxMemoryPool> default_pool =
    std::make_shared<WrappedVeloxMemoryPool>(...);
```

Reply:
Thanks, use static std::shared_ptr<WrappedVeloxMemoryPool> can fix core.